api/dao/liststorage.py

```diff
@@ -123,15 +113,48 @@ def _delete_el(self, _id, query_params):
         return result
 
     def _get_el(self, _id, query_params):
-        log.debug('query_params {}'.format(query_params))
         query = {'_id': _id, self.list_name: {'$elemMatch': query_params}}
         projection = {self.list_name + '.$': 1}
-        log.debug('query {}'.format(query))
-        log.debug('projection {}'.format(projection))
         result = self.dbc.find_one(query, projection)
         if result and result.get(self.list_name):
             return result.get(self.list_name)[0]
 
+    def modify_info(self, _id, query_params, payload):
+        update = {}
+        set_payload = payload.get('set')
+        delete_payload = payload.get('delete')
+        replace_payload = payload.get('replace')
```

Comment:
Kofalt and I briefly discussed this. Since it's not business or schema logic, I added an extra check for other parts of the code that may call this function. I'd rather have a descriptive 500 if someone misused it than a mongo error about key-wise vs full key sets. I do agree that business logic/schema validation belongs at the handler levels.